src/cuda-sim/ptx_sim.cc

```diff
@@ -369,7 +369,7 @@ static void print_reg(FILE *fp, std::string name, ptx_reg_t value,
       fprintf(fp, ".u64 %llu [0x%llx]\n", value.u64, value.u64);
       break;
     case F16_TYPE:
-      fprintf(fp, ".f16 %f [0x%04x]\n", value.f16, (unsigned)value.u16);
+      fprintf(fp, ".f16 %f [0x%04x]\n", static_cast<float>(value.f16), (unsigned)value.u16);
```

Comment:
Please use method `half::operator float()` to do the conversion.

Reply:
Please disregard the above comment. Your code is correct, static_cast will call the overloaded float typecast of `half`.